dfastmi/gui/dialog_view_model.py

```diff
@@ -0,0 +1,296 @@
+            time_fractions_of_the_year = ConfigurationInitializer.get_time_fractions_of_the_year(self._current_reach.hydro_t)
+            time_mi = ConfigurationInitializer.calculate_time_mi(q_threshold, self._current_reach.hydro_q, time_fractions_of_the_year)
+            celerity = ConfigurationInitializer.get_bed_celerity(self._current_reach, self._current_reach.hydro_q)
+            slength = dfastmi.kernel.core.estimate_sedimentation_length(time_mi, celerity)
+            self.slength = "{:.0f}".format(slength)
+        except:
+            self.slength = "---"
+    
+    def save_configuration(self, filename: str) -> None:
+        """
+        Open a configuration file and update the GUI accordingly.
+
+        This routines opens the specified configuration file and updates the GUI
+        to reflect it contents.
+
+        Arguments
+        ---------
+        filename : str
+            Name of the configuration file to be saved.
+            
+        """
+        config = self.model.get_configuration()
+        ConfigFileOperations.save_configuration_file(filename, config)
+
+    def load_configuration(self, filename: str) -> bool:
+        """
+        Open a configuration file and update the GUI accordingly.
+
+        This routines opens the specified configuration file and updates the GUI
+        to reflect it contents.
+
+        Arguments
+        ---------
+        filename : str
+            Name of the configuration file to be opened.
+        """
+        self.model.load_configuration(filename)
+        
+        self._output_dir = self.model.section.get("OutputDir", "output")
+        self._plotting = self.str_to_bool(self.model.section.get("Plotting", "false"))
+        self._save_plots = self.str_to_bool(self.model.section.get("SavePlots", "false"))
+        self._close_plots = self.str_to_bool(self.model.section.get("ClosePlots", "false"))
+
+        
+        self.update_qvalues()
+        
+        hydro_q = self._current_reach.hydro_q
+        for i in range(len(hydro_q)):
+            prefix = str(i)+"_"
+            cond = "C{}".format(i+1)
+            if cond in self.model.config.keys():
+                cond_section = self.model.config[cond]
+                self._reference_files.append(cond_section.get("Reference", ""))
+                self._measure_files.append(cond_section.get("WithMeasure", ""))
+            else:
+                self._reference_files.append("")
+                self._measure_files.append("")
+        
+        self.current_branch = self.model.rivers.get_branch(self.model.section["Branch"])
+        self.current_reach = self.current_branch.get_reach(self.model.section["Reach"])
+        
+        return True
+    
+    def check_configuration(self) -> bool :
+        return self.model.check_configuration(self.current_branch, self.current_reach, self.reference_files, self.measure_files)
+        
+    def str_to_bool(self, x: str) -> bool:
```

Comment:
this will be replaced with changes from DFAST-145?

Reply:
yup. did not want to change too much.